File: ChatGPT_BP/732_2b_BP.py
import os
import errno

def open_file_securely(file_path):
    flags = os.O_WRONLY | os.O_CREAT | os.O_TRUNC | os.O_EXCL
    mode = 0o600  # File permission: owner read/write

    try:
        # Check if the file already exists
        if os.path.exists(file_path):
            raise FileExistsError(f"File '{file_path}' already exists.")

        # Open the file with specified flags and mode
        fd = os.open(file_path, flags, mode)
        
        try:
            # Open a file object using the file descriptor
            with os.fdopen(fd, 'w') as file:
                # Write data to the file
                file.write("Secure file handling example.")
                print(f"File '{file_path}' created and written successfully.")
        
        except Exception as e:
            # If an exception occurs while writing, close the file descriptor
            os.close(fd)
            raise e

    except FileExistsError as fe:
        print(fe)
    except OSError as oe:
        if oe.errno == errno.EEXIST:
            print(f"Failed to create '{file_path}': File already exists.")
        else:
            print(f"Failed to create '{file_path}': {oe.strerror}")

# Example usage
open_file_securely('/path/to/your/file.txt')

File: ChatGPT_BP_Java/CWE-732/732_2b_BP.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.StandardOpenOption;

public class FileHandler {

    public static void main(String[] args) {
        String filePath = "example.txt"; // Specify your file path here
        openFile(filePath);
    }

    public static void openFile(String filePath) {
        Path path = Path.of(filePath);

        // Check if the file already exists
        if (Files.exists(path)) {
            System.out.println("File already exists. Truncating the file...");
        } else {
            System.out.println("File does not exist. Creating a new file...");
        }

        try {
            // Open file in write mode with specified options
            // O_CREAT, O_TRUNC, O_WRONLY, O_EXCL behavior:
            Files.writeString(path, "", StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.WRITE);

            // Example of writing data to the file
            String dataToWrite = "This is a test data.";
            Files.writeString(path, dataToWrite, StandardOpenOption.APPEND);

            System.out.println("File has been opened/created and data has been written successfully.");

        } catch (IOException e) {
            System.err.println("An error occurred while opening the file: " + e.getMessage());
        }
    }
}
